oxen/src/py_remote_repo.rs

```diff
@@ -95,18 +108,118 @@ impl PyRemoteRepo {
+        let user_id = UserConfig::identifier()?;
+        pyo3_asyncio::tokio::get_runtime().block_on(async {
+            api::remote::staging::add_file(
+                &self.repo,
+                &self.revision,
+                &user_id,
+                &directory_name,
+                path,
+            )
+            .await
+        })?;
+        Ok(())
+    }
+
+    fn remove(&self, path: PathBuf) -> Result<(), PyOxenError> {
+        let user_id = UserConfig::identifier()?;
+        pyo3_asyncio::tokio::get_runtime().block_on(async {
+            api::remote::staging::rm_staged_file(&self.repo, &self.revision, &user_id, path).await
+        })?;
+        Ok(())
+    }
+
+    fn commit(&self, message: String) -> Result<(), PyOxenError> {
+        let user_id = UserConfig::identifier()?;
+        let user = UserConfig::get()?.to_user();
+        let commit = CommitBody { message, user };
+        pyo3_asyncio::tokio::get_runtime().block_on(async {
+            api::remote::staging::commit_staged(&self.repo, &self.revision, &user_id, &commit).await
+        })?;
+        Ok(())
+    }
+
+    fn log(&self) -> Result<Vec<PyCommit>, PyOxenError> {
+        let log = pyo3_asyncio::tokio::get_runtime().block_on(async {
+            api::remote::commits::list_commit_history(&self.repo, &self.revision).await
+        })?;
+        Ok(log.iter().map(|c| PyCommit { commit: c.clone() }).collect())
+    }
+
+    fn add_df_row(&self, path: PathBuf, data: String) -> Result<(), PyOxenError> {
+        let user_id = UserConfig::identifier()?;
+        pyo3_asyncio::tokio::get_runtime().block_on(async {
+            api::remote::staging::stage_modification(
+                &self.repo,
+                &self.revision,
+                &user_id,
+                &path,
+                data,
+                ContentType::Json,
+                ModType::Append,
+            )
+            .await
+        })?;
+        Ok(())
+    }
+
+    fn list_branches(&self) -> Result<Vec<PyBranch>, PyOxenError> {
+        let branches = pyo3_asyncio::tokio::get_runtime()
+            .block_on(async { api::remote::branches::list(&self.repo).await })?;
+        Ok(branches
+            .iter()
+            .map(|b| PyBranch {
+                name: b.name.clone(),
+                commit_id: b.commit_id.clone(),
+            })
+            .collect())
+    }
+
+    fn status(&self, path: PathBuf) -> Result<PyStagedData, PyOxenError> {
+        let user_id = UserConfig::identifier()?;
+        let remote_status = pyo3_asyncio::tokio::get_runtime().block_on(async {
+            api::remote::staging::status(
+                &self.repo,
+                &self.revision,
+                &user_id,
+                &path,
+                liboxen::constants::DEFAULT_PAGE_NUM,
+                liboxen::constants::DEFAULT_PAGE_SIZE,
+            )
+            .await
+        })?;
+
+        let mut status = StagedData::empty();
```

Comment:
It feels like we should do this in a From trait on the PyStagedData struct

Reply:
done!

Reply:
beautiful